src/downloader/downloader.cpp

```diff
@@ -5,34 +5,195 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+
 #include "downloader.hpp"
+#include "log.hpp"
+
+Downloader::Downloader()
+    : mDownloadRequester(nullptr)
+    , mThread()
+    , mMutex()
+    , mWaitDownload(mMutex)
+    , mErrProcessImageRequest()
+    , mRequestedPath()
+    , mDownloadResults()
+    , mTimer(Downloader::DownloadTimeoutCallback, this)
+    , mRequestID(0)
+{
+}
+
+Downloader::~Downloader()
+{
+    mWaitDownload.NotifyOne();
```

Comment:
This is useless if you don't wait until notified thread is really finished.
Also constructor will be never called as this app never finishes.